.github/workflows/build.yml

```
@@ -62,29 +62,86 @@ jobs:
     runs-on: ubuntu-20.04
     steps:
       - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
-      - uses: redhat-actions/buildah-build@d097e2e3d2a45b68a7c09040e3de33427bab66af
-        id: build-image
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to Quay
+        uses: docker/login-action@v2
         with:
-          image: build
-          oci: true
-          dockerfiles: |
-            Dockerfile.build-image
-      - if: github.ref == 'refs/heads/main'
-        uses: redhat-actions/push-to-registry@df878026e36864aa6bf9a5b7afe63637cb88ef8c
-        with:
-          image: ${{ steps.build-image.outputs.image }}
           registry: quay.io/security-profiles-operator
           username: security-profiles-operator+github
           password: ${{ secrets.QUAY_TOKEN }}
+        if: github.ref == 'refs/heads/main'
+      - name: Image metadata
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            quay.io/security-profiles-operator/security-profiles-operator
+          tags: |
+            type=ref,event=branch
+            type=sha,format=long
+      - name: Build (and push if needed)
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: Dockerfile.build-image
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          push: ${{ github.ref == 'refs/heads/main' }}
+          load: true
+      - name: Run container image vulnerability scanner
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: ${{ steps.meta.outputs.tags }}
+          format: 'table'
+          # TODO(jaosorior): Fix this once we bump the NixOS dependencies.
+          exit-code: '0'
+          ignore-unfixed: true
+          vuln-type: 'os,library'
+          severity: 'CRITICAL,HIGH'
+          timeout: 10m
 
   ubi-image:
     runs-on: ubuntu-20.04
     steps:
       - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
-      - uses: redhat-actions/buildah-build@d097e2e3d2a45b68a7c09040e3de33427bab66af
-        id: build-image
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
```

I think for most things we are using the hash commit as version:
```suggestion
        uses: docker/setup-buildx-action@312f67c6629946ba17269558b63d0bd84a63caab
```